core/src/main/scala/org/apache/spark/util/AsynchronousListenerBus.scala

```diff
@@ -66,6 +66,7 @@ private[spark] abstract class AsynchronousListenerBus[L <: AnyRef, E](name: Stri
         self.synchronized {
           processingEvent = true
         }
+        if (stopped.get()) return
```

Comment:
instead of doing this here, can you simplify the try block as follows?
```
try {
  if (stopped.get()) { return }
  val event = eventQueue.poll()
  assert(event != null, "event queue was empty but the listener bus was not stopped")
  postToAll(event)
} finally {
```